src/compiler/types.rs

```diff
@@ -184,32 +176,30 @@ impl NodeInterface for BaseNode {
         *self.symbol.borrow_mut() = Some(Rc::downgrade(&symbol));
     }
 
-    fn locals(&self) -> MappedRwLockWriteGuard<SymbolTable> {
-        RwLockWriteGuard::map(self.locals.try_write().unwrap(), |option| {
-            option.as_mut().unwrap()
-        })
+    fn locals(&self) -> RefMut<SymbolTable> {
+        RefMut::map(self.locals.borrow_mut(), |option| option.as_mut().unwrap())
```

Comment:
`RefCell`'s support `map()`'ing the borrowed contents, `Ref::map()`/`RefMut::map()` just return the same type as `.borrow()`/`.borrow_mut()` (`Ref`/`RefMut`)